### PR TITLE
ENH add new stackvana build up to meas_algorithms

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge/staged-recipes This PR is ready for review. 

One important note is that I've turned off overlinking errors. The recipe here is buildind software from the Vera C. Rubin Observatory. The Rubin Observatory uses a specialized build susytem that merges build, host, and run into a single environment. It further relies on a metapackage (rubin-env) plus the EUPS build tool to define its own custom ABI. The stackvana-core package pulls in this environment definition and appears in both build/host and run. Conda doesn't see the explicit requirements for the linkages it finds and so complains. 

Before we merge, we should turn the overlinking errors back on.

**Comment by isuruf** [comment]
Unless stackvana-core pins its dependencies and transitive dependencies exactly (by exactly I mean pinning the exact hash string) and `stackvana-core` here is pinned exactly, removing overlinking error is wrong.

**Comment by beckermr** [comment]
Yes those pins are in Rubin-env.

**Comment by beckermr** [comment]
@conda-forge/staged-recipes shall we merge?

**Comment by isuruf** [comment]
I don't see any tight pinning in https://github.com/conda-forge/rubinenv-feedstock/blob/main/recipe/meta.yaml. So, I'm -1 to merging as is.

**Comment by beckermr** [comment]
We pin using a combination of .* past the global pin plus run exports from host. This pins things to be greater than or equal to the build version but less than the next possibly compatible version in run exports. 

We've been shipping a full stack of packages like this for months now and have had no problems.

**Comment by beckermr** [comment]
Any other comments here @conda-forge/staged-recipes or shall we merge?

File: recipes/stackvana-meas_algorithms/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    - stackvana-core =={{ version }}
+    - stackvana-afw =={{ version }}
+
+test:
+  commands:
+    - setup meas_algorithms
+    - python -c "import lsst.afw"
+
+about:
+  home: https://github.com/beckermr/stackvana-core
```
Comment (carterbox) [inline recipes/stackvana-meas_algorithms/meta.yaml:47]: It's hard for me to review this recipe when this "home" page is not a public repository, and the source download is just a plain text list of packages?

Comment (beckermr) [reply recipes/stackvana-meas_algorithms/meta.yaml:47]: The source is public. The list of packages is a set of eups requirements that is fed to EUPS and built.

**Comment by beckermr** [comment]
Just for reference, we already a ship a bunch of feedstocks like this: 

https://github.com/conda-forge/stackvana-core-feedstock
https://github.com/conda-forge/stackvana-afw-feedstock
https://github.com/conda-forge/stackvana-feedstock

These feedstocks wrap specialized software for the Vera C. Rubin Observatory. Their software uses the EUPS package manager (https://github.com/RobertLuptonTheGood/eups) to install custom C++ / python packages. They use calver and have no stable ABIs / APIs. Thus separating out the distinct EUPS packages into conda packages is not feasible or practical. Instead, the packages here on conda-forge wrap the EUPS build within conda, using activation scripts to setup EUPS. The feedstocks above are separate parts of the build and exist to ensure the build can complete in CI time requirements. 

I am adding this apckage (which is in the middle between afw and the final lsstdistrib build) because the CI has been running out of time.

File: recipes/stackvana-meas_algorithms/meta.yaml (hunk 1 above)
Comment (beckermr) [reply recipes/stackvana-meas_algorithms/meta.yaml:47]: The versions are defined by this list which is why that list is the source. This list gets updated which is when you know a new version exists. This triggers the bot.

**Comment by beckermr** [comment]
@conda-forge/staged-recipes Any other comments / questions here or can we merge?

**Review by carterbox — CHANGES_REQUESTED** [review]
The link checker states that runtime dependencies such as fftw, mkl, and the c runtimes are missing from the dependency list. I'm not sure why you would want to ignore errors about these missing deps.

There's also a whole bunch of clobber warnings between this package and idds-client.

**Comment by beckermr** [comment]
See the discussion with @isuruf above. These deps are brought in by the rubin-env package.

**Comment by beckermr** [comment]
The clobber warnings are unavoidable AFAICT. 

The reason we don't repeat the host deps everywhere is that there are multiple feedstocks and it is a huge pain to copy around the correct pins in all the places.

**Comment by beckermr** [comment]
I'm honestly very confused. We've already merged PRs like this for feedstocks in the past. This software is used by me and my collaborators and has been running fine for over a year. IDK why suddenly this is being blocked.

**Comment by carterbox** [comment]
> I'm honestly very confused.

I also feel confused because this is an irregular package which doesn't follow the conventions of a majority of the software on the channel. 

> We've already merged PRs like this for feedstocks in the past. ... IDK why suddenly this is being blocked.

Sorry that the pushback is unexpected to you. However, it looks like only yourself and @ocefpaf have reviewed the related packages you have mentioned. Perhaps, @ocefpaf can give you the review that you were expecting.

> The reason we don't repeat the host deps everywhere is that there are multiple feedstocks and it is a huge pain to copy around the correct pins in all the places.

You could put all the pins into a `conda_build_config.yaml`, then you just copy the file around instead of manually editing the text in each feedstock?

**Comment by beckermr** [comment]
I understand, but given my own expertise on shipping packages in conda-forge, the fact that this is used by a small number of people for a very custom use case (and so won't effect the broader community), and that it has been working fine, I was hoping for the benefit of the doubt.

In terms of the pins, we already use a conda build config here (https://github.com/conda-forge/rubinenv-feedstock/blob/main/recipe/conda_build_config.yam) that is used to build rubin-env. We'd have to recopy it to three other feedstocks once a week as it gets changed which is a pain. The unusual packaging pattern here let's us avoid that.

**Comment by carterbox** [comment]
If the `rubinenv` package contains all the pins, then shouldn't this recipe and the others all depend directly on a specific version of the `rubinenv` package?

**Comment by beckermr** [comment]
Yep. The stackvana-core package has that dep pinned to the correct version.

**Comment by beckermr** [comment]
@carterbox You ok if we merge now?

**Review by carterbox — APPROVED** [review]
I see that rubin-env-noroot is pinned to the major version. Like Isuru, I am skeptical that the pinnings are strict enough, but I believe that any impact of breakage will be limited to your specific community. I approve with reservations. As you like to say, "Let perfect not be the enemy of good".

File: .ci_support/build_all.py [hunk 2]
```diff
@@ -133,7 +133,7 @@ def get_config(arch, channel_urls):
 
     config = conda_build.api.Config(
         arch=arch, exclusive_config_files=exclusive_config_files,
-        channel_urls=channel_urls, error_overlinking=True,
+        channel_urls=channel_urls, error_overlinking=False,
```
Comment (carterbox) [inline .ci_support/build_all.py:136]: Please remember to revert these changes

Comment (beckermr) [inline .ci_support/build_all.py:136]:
```suggestion
        channel_urls=channel_urls, error_overlinking=True,
```

**Comment by beckermr** [comment]
Thank you everyone for your patience with this PR!
